internal/role/p2p/k8s.go

```diff
@@ -0,0 +1,57 @@
+package role
+
+import (
+	"errors"
+
+	"github.com/kairos-io/kairos-sdk/machine"
+	"github.com/kairos-io/kairos-sdk/utils"
+	providerConfig "github.com/kairos-io/provider-kairos/v2/internal/provider/config"
+	service "github.com/mudler/edgevpn/api/client/service"
+)
+
+type K8sNode interface {
+	propagateData() error
```

Comment:
non exported function in interface? That's a strange requirement. Since nothing except the instance itself has access to this method, there is no reason to require it in the interface no?

Reply:
yeah it was already existing and I was a bit too lazy to refactor that too :P ... will only export for now, and later on we can improve the refactoring if needed